foam_controlwrapper/tests/test_controlwrapper.py

```diff
@@ -6,66 +6,266 @@
     """Test case for FoamControlWrapper class"""
     def setUp(self):
-        pass
+        self.mesh = Mesh(name="mesh1")
+
+        self.points = [
+            Point(
+                (0.0, 0.0, 0.0)),
+            Point(
+                (1.0, 0.0, 0.0)),
+            Point(
+                (1.0, 1.0, 0.0)),
+            Point(
+                (0.0, 1.0, 0.0)),
+            Point(
+                (0.0, 0.0, 1.0)),
+            Point(
+                (1.0, 0.0, 1.0)),
+            Point(
+                (1.0, 1.0, 1.0)),
+            Point(
+                (0.0, 1.0, 1.0))
+        ]
+
+        puids = [self.mesh.add_point(point) for point in self.points]
+
+        self.faces = [
+            Face([puids[0], puids[3], puids[7], puids[4]],
+                 data=DataContainer({CUBA.LABEL: 0})),
+            Face([puids[1], puids[2], puids[6], puids[5]],
+                 data=DataContainer({CUBA.LABEL: 1})),
+            Face([puids[0], puids[1], puids[5], puids[4]],
+                 data=DataContainer({CUBA.LABEL: 2})),
+            Face([puids[3], puids[2], puids[6], puids[7]],
+                 data=DataContainer({CUBA.LABEL: 3})),
+            Face([puids[0], puids[1], puids[2], puids[3]],
+                 data=DataContainer({CUBA.LABEL: 4})),
+            Face([puids[4], puids[5], puids[6], puids[7]],
+                 data=DataContainer({CUBA.LABEL: 5}))
+
+        ]
+
+        [self.mesh.add_face(face) for face in self.faces]
+
+        self.cells = [
+            Cell(puids)
+        ]
+
+        self.puids = puids
+
+        [self.mesh.add_cell(cell) for cell in self.cells]
+
+    def tearDown(self):
+        FoamFiles().remove_parser_files(os.getcwd())
+
+    def test_add_mesh(self):
+        """Test add_mesh method
+
+        """
+
+        wrapper = FoamControlWrapper()
+        wrapper.add_mesh(self.mesh)
+
+    def test_delete_mesh(self):
+        """Test delete_mesh method
+
+        """
+
+        wrapper = FoamControlWrapper()
+        wrapper.add_mesh(self.mesh)
+        wrapper.delete_mesh(self.mesh.name)
+
+    def test_get_mesh(self):
+        """Test get_mesh method
+
+        """
+
+        wrapper = FoamControlWrapper()
+        wrapper.add_mesh(self.mesh)
+        mesh_inside_wrapper = wrapper.get_mesh(self.mesh.name)
+        self.assertEqual(self.mesh.name, mesh_inside_wrapper.name)
+
+        for point in self.mesh.iter_points():
+            point_w = mesh_inside_wrapper.get_point(point.uid)
+            self.assertEqual(point.uid, point_w.uid)
+            self.assertEqual(point.coordinates, point_w.coordinates)
+
+        for face in self.mesh.iter_faces():
+            face_w = mesh_inside_wrapper.get_face(face.uid)
+            self.assertEqual(face.uid, face_w.uid)
+            self.assertEqual(face.points, face_w.points)
+
+        for cell in self.mesh.iter_cells():
+            cell_w = mesh_inside_wrapper.get_cell(cell.uid)
+            self.assertEqual(cell.uid, cell_w.uid)
```

Comment:
in the code, it looks like `cell.data` is not empty; the cell.data has velocity and pressure.  could such values be added to the tested cells here.